samples/debugmalloc/README.md

```diff
@@ -0,0 +1,328 @@
+  "OE_UNEXPECTED".
+
+- `    oe_result_t oe_debug_malloc_tracking_report(
+        uint64_t* out_object_count,
+        char** report)
+  `
+
+  Report the leaks, including the number and details of allocations that have
+  not been freed since start was called. If called after stop, it reports the
+  set of allocations that were not freed while tracking is turned on, even if
+  they might have been freed between the calls to stop and report. If called
+  before the earliest start, the report includes nothing.
+
+Please note that the timing and sequences of calling of these functions is
+important:
+
+- The starting and stopping tracking must cover the coupled malloc/free
+operations, which means start tracking before malloc and also stop tracking
+after free, or the report will definitely trigger false alarms that some
+reported objects actually are freed after stopping tracking.
+
+- oe_debug_malloc_tracking_stop() can only been called after calling
+oe_debug_malloc_tracking_start(), and oe_debug_malloc_tracking_start() can be
+called at the beginning or after calling oe_debug_malloc_tracking_stop(). While
+the calling sequence following this rule, the return values are OE_OK. Return
+value "OE_UNEXPECTED" means the sequence is not expected.
+
+Regardless the execution and output of these state switching functions,
+oe_debug_malloc_tracking_report() covers all the leaks happened during the
+tracking state cumulatively.
+
+```
+      stop()  |--------------|  start() |--------------|  start()
+   |--------->|              |--------->|              |----------|
+   |          | not tracking |          |   tracking   |          |
+   |----------|              |<---------|              |<---------|
+              |--------------|  stop()  |--------------|
+```
+
+- oe_debug_malloc_tracking_start() and oe_debug_malloc_tracking_stop() can be
+called many times:
+
+```
+  |----code1--|-----------|----code2--|--|----code3--|
+  start1->stop1->report1->start2->stop2->start3->stop3->report3
+```
+
+In this case report1 only covers code1, and report3 covers code1, code2 and
+code3.
+
+Please also note that these public APIs only affect the result of local
+detection. The result of global detection is independent of these APIs. While
+the calling of these APIs follow the rules, the result of local detection is
+supposed to be the subset of result of global detection.
+
+## Example
+```c
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/debugmalloc.h>
+#include <stdio.h>
+
+// Include the trusted debugmalloc header that is generated
+// during the build. This file is generated by calling the
+// sdk tool oeedger8r against the debugmalloc.edl file.
+#include "debugmalloc_t.h"
+
+// This is the function that the host calls. It prints
+// a message in the enclave before calling back out to
+// the host to print a message from there too.
+void enclave_hello()
+{
+    void *buf1, *buf2, *buf3;
+    uint64_t num;
```

Comment:
See comments in the actual sample.  The same changes are needed here.

Reply:
num -> count
buf->buffer

Reply:
Updated, thanks!